modules/menu/static_menu.c

```diff
@@ -158,10 +158,11 @@ static int cmd_answerdir(struct re_printf *pf, void *arg)
 		ua = call_get_ua(call);
 	}
 
-	if (call_sent_answer(call))
-		(void)call_set_media_estdir(call, adir, vdir);
-	else
-		(void)call_set_media_direction(call, adir, vdir);
+	(void)call_set_media_estdir(call, adir, vdir);
+
+	enum sdp_neg_state sdp_state = call_sdp_neg_state(call);
+	if (sdp_state == SDP_NEG_NONE || sdp_state == SDP_NEG_REMOTE_OFFER)
```

Comment:
Is it possible to move this checks to the baresip core?

Reply:
If a change applies to all apps and not just the menu app, I would prefer to make the change in the core so that all apps would benefit.

Reply:
I added the `call_sdp_change_allowed` function to call.c.

I can not move this check into the `call_set_media_direction` function because this function **only** sets the local directions of the streams and this could for example break the `/medialdir` command in `dynamic_menu.c`.